Automattic/jetpack
# Site Importer: Custom Experience for action: jetpack_import_ui

#### Changes proposed in this Pull Request:


* New (auto-run) module `site-importer/site-importer.php`
* Hook into `admin_action_jetpack_import_ui`
* If on the importer page, replace the UI (currently a rudimentary WIP)
* Provide an "exit" button to show the default UI

#### Testing instructions:

* Go to `/wp-admin/import.php?action=jetpack_import_ui`
  * You should see a (WIP) custom UI
  * Clicking the `Exit` button should show the "default" UI & all flows should work as before
* Go to `/wp-admin/import.php`
  * You should see the "default" UI & all flows should work as before
* Go to other wp-admin pages (with and without the `action=jetpack_import_ui` query argument
  * They should not be affected
* Frontend page loads should not be affected
* AJAX calls should not be affected
* JSON API calls should not be affected

#### Proposed changelog entry for your changes:

* `@TODO`
